### ISSUE TITLE
How are Python classes / functions serialized / deserialized?

### ISSUE THREAD

**Issue by vadimkantorov** [joblib/joblib#1619]
Are they stored as Python bytecode?

A bit missing this info at https://joblib.readthedocs.io/en/stable/persistence.html

Is joblib equivalent to pickle in this regard?

We observed function speed-ups after `joblib.dump`/`joblib.load`-ing which is counter-intuitive and so trying to understand what's going on and if overhead in running bare Python code and `joblib.load`-ed function is different...

**Comment by tomMoral** [comment]
For functions, this should be the same as `pickle.dump` so I don't see why it would be faster.
If you have a reproducer, we can try to investigate this.

**Comment by Spill-Tea** [comment]
@tomMoral Doesn't the loky backend use cloudpickle?

**Comment by tomMoral** [comment]
yes, but this is not used by `joblib.dump` so I don't think it explains the performance boost from this issue

**Comment by vadimkantorov** [comment]
We observed that using pickle instead eliminates the boost. Will try to post a repro here.
